checks/evaluation/branch_protection.go

```diff
@@ -362,6 +328,18 @@ func adminReviewProtection(branch *clients.BranchRef, dl checker.DetailLogger) (
 		}
 	}
 
+	max++
+	if branch.BranchProtectionRule.RequiredPullRequestReviews != nil {
+		score++
+		info(dl, log, "PRs are required in order to make changes on branch '%s'", *branch.Name)
+	} else {
+		warn(dl, log, "PRs are not required to make changes on branch '%s'; or we don't have data to detect it."+
+			"If you think it might be the latter, make sure to run Scorecard with a PAT or use Repo "+
+			"Rules (that are always public) instead of Branch Protection settings", *branch.Name)
+		// Warning it here because since `RequiredPullRequestReviews` is nil, we won't check reviewers count afterwards
+		warn(dl, log, "No reviewers required to merge changes on branch '%s'", *branch.Name)
```

Comment:
Actually I think it's working as I intended, but let me know if you'd rather change it (or add clarifying comments). My strategy is:
- For the `adminReviewProtection` function(code below), I'm treating only the case of **no PRs required** (i.e., a nil RequiredPullRequestReviews struct). Then, I'm trowing a warning complaining that the branch doesn't require PRs, but I'm also trowing another warning complaining about having no reviewers, because we want the maintainer to require PRs but we also want them to require reviewers
```
max++
	if branch.BranchProtectionRule.RequiredPullRequestReviews != nil {
		score++
		info(dl, log, "PRs are required in order to make changes on branch '%s'", *branch.Name)
	} else {
		warn(dl, log, "PRs are not required to make changes on branch '%s'; or we don't have data to detect it."+
			"If you think it might be the latter, make sure to run Scorecard with a PAT or use Repo "+
			"Rules (that are always public) instead of Branch Protection settings", *branch.Name)
		// Warning it here because since `RequiredPullRequestReviews` is nil, we won't check reviewers count afterwards
		warn(dl, log, "No reviewers required to merge changes on branch '%s'", *branch.Name)
	}
```
- For the `nonAdminThoroughReviewProtection` , we exclude the case where no PRs are required, so there shouldn't be duplications of warning on the case of no PRs required. This function will only emit loggings if the branch requires PRs.
```
	// On this first check we exclude the case where PRs are not required to make changes,
	// because it's already covered on adminReviewProtection function.
	if branch.BranchProtectionRule.RequiredPullRequestReviews != nil &&
		branch.BranchProtectionRule.RequiredPullRequestReviews.RequiredApprovingReviewCount != nil {
		if *branch.BranchProtectionRule.RequiredPullRequestReviews.RequiredApprovingReviewCount >= minReviews {
			info(dl, log, "number of required reviewers is %d on branch '%s'",
				*branch.BranchProtectionRule.RequiredPullRequestReviews.RequiredApprovingReviewCount, *branch.Name)
			score++
		} else {
			warn(dl, log, "number of required reviewers is %d on branch '%s', while the ideal suggested is %d",
				*branch.BranchProtectionRule.RequiredPullRequestReviews.RequiredApprovingReviewCount, *branch.Name, minReviews)
		}
	}
```

Answering your question number 2, for the case of review count = 0, the `adminReviewProtection` won't trigger any warnings, indeed, it would only emit the "info" that the branch requires PRs. The warning about the number of reviewers would be triggered on the `nonAdminThoroughReviewProtection` function.

Reply:
> Actually I think it's working as I intended, but let me know if you'd rather change it

I feel like I had some case last Friday, but it was also late on a Friday so maybe I was mistaken.